src/binary/ibc-setup/utils/load-and-validate-app.ts

```diff
@@ -0,0 +1,36 @@
+import fs from 'fs';
+import path from 'path';
+
+import Ajv, { JSONSchemaType } from 'ajv';
+import yaml from 'js-yaml';
+
+import { appFile } from '../../constants';
+import { App } from '../types';
+
+export function loadAndValidateApp(home: string) {
+  try {
+    const app = yaml.load(fs.readFileSync(path.join(home, appFile), 'utf-8'));
+
+    const ajv = new Ajv({ allErrors: true });
+    const schema: JSONSchemaType<App> = {
+      type: 'object',
+      additionalProperties: false,
+      required: [],
+      properties: {
+        src: { type: 'string', nullable: true, default: null },
+        dest: { type: 'string', nullable: true },
+        mnemonic: { type: 'string', nullable: true },
+        keyFile: { type: 'string', nullable: true },
+      },
+    };
+    const validate = ajv.compile(schema);
+
+    if (!validate(app)) {
+      return null;
+    }
+
+    return app;
+  } catch {
+    return null;
+  }
+}
```

Comment:
The `app.yaml` is optional, thus this function return type is `App | null`. Does it make sense? @ethanfrey

Reply:
Yes.

Reply:
I would only catch errors around the missing file (read errors), and return null or undefined if missing.

The validation error should throw, so people know why the values in the file are not being used

Reply:
00fb34f